### ISSUE TITLE
Make disabling linting in macro calls "easier" at the cost of false negatives

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#2010]
Some people who are just getting started with clj-kondo or are using it on a big codebase with lots of custom macros may could use an easy way to disable clj-kondo false positives in macro calls at the cost of false negatives.

Let's take meander as an example:

``` clojure
(require '[meander.epsilon :as m])


(defn favorite-food-info [foods-by-name user]
  (m/match {:user user
            :foods-by-name foods-by-name}
    {:user
     {:name ?name
      :favorite-food {:name ?food}}
     :foods-by-name {?food {:popularity ?popularity
                            :calories ?calories}}}
    {:name ?name
     :favorite {:food ?food
                :popularity ?popularity
                :calories ?calories}}))
```

Although it may be possible to write a hook to implement correct analysis with respect to locals like `?food`, it's can be daunting at first. (A similar hook can be found [here](https://github.com/metabase/metabase/blob/8e102664094be7d7133b34d7439f991faac27a33/.clj-kondo/hooks/metabase/test/data.clj#L83) which is quite involved).

We already have similar rules for `def`-like macros: `{:lint-as {my-macro/def clj-kondo.lint-as/def-catch-all}}` recognizes the first symbol after the `def`-like symbol and then ignores all the rest of the body.

We could introduce a `{:lint-as {meander.epsilon/match clj-kondo.lint-as/ignored}}` which disables (most) linters within calls to the macro.

To further ease clj-kondo usage, we could also support an inline annotation on the macro metadata:

``` clojure
(defmacro match
  {clj-kondo/ignore true}
  [...] ...)
```

or to only suppress certain linters in the macro call:

``` clojure
(defmacro match
  {clj-kondo/ignore [:unresolved-symbol]}
  [...] ...)
```

cc @noprompt
